pkg/controller/direct/registry/registry.go

```diff
@@ -32,6 +33,36 @@ type registry struct {
 	registrations map[schema.GroupKind]*registration
 }
 
+// staticRegistrations contain all the GKs and their highest versions supported
+// via direct controllers.
+// It's the static set of the direct controllers, and it should contain all the
+// GKs NOT based on TF, DCL, or handwritten IAM types.
+var staticRegistrations = map[schema.GroupKind]string{
```

Comment:
Not sure I understand the purpose of this mapping. Don't the direct controllers reigster themselves with `RegisterModel` in their `init()` functions already? Can we use their registrations instead of these static ones?

Reply:
The registration via init() doesn't happen in unit tests or local scripts. I have discussed it with Yuwen and decided to go with a different route: https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2664#discussion_r1755254382